### PR TITLE
Test python 3.13

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m venv venv
-          venv/bin/pip install -r requirements.txt
+          venv/bin/pip install --no-deps --only-binary ':all:' -r requirements.txt
```
Comment (willkg) [inline .github/workflows/build.yaml:28]: Out of curiosity, what does this have to do with supporting Python 3.13?

Comment (willkg) [reply .github/workflows/build.yaml:28]: ubernostrum suggested this in his recent blog post:

https://www.b-list.org/weblog/2023/dec/07/pip-install-safely/

I tried it out with and had issues with some packages not having wheels and thus being uninstallable. I don't know if I have any evidence of that anywhere and even if I did, it's probably worth trying again.

Comment (relud) [reply .github/workflows/build.yaml:28]: this step was failing for python 3.13 because it couldn't build `grpcio`. I don't think it should be trying to build that package, so i added this to see if it would make a difference. it didn't, so i'm going to close this PR for now.

Comment (willkg) [reply .github/workflows/build.yaml:28]: Ahhh... yuck.
